Dockerfile

```diff
@@ -1,32 +1,26 @@
-ARG base_image=ghcr.io/alphagov/govuk-ruby-base:3.1.2
-ARG builder_image=ghcr.io/alphagov/govuk-ruby-builder:3.1.2
+ARG ruby_version=3.1.2
+ARG base_image=ghcr.io/alphagov/govuk-ruby-base:$ruby_version
+ARG builder_image=ghcr.io/alphagov/govuk-ruby-builder:$ruby_version
 
-FROM $builder_image AS builder
-
-RUN mkdir -p /app && ln -fs /tmp /app/tmp && ln -fs /tmp /home/app
 
-WORKDIR /app
-
-COPY Gemfile* .ruby-version /app/
+FROM $builder_image AS builder
 
+WORKDIR $APP_HOME
+COPY Gemfile* .ruby-version ./
 RUN bundle install
-
-COPY . /app
-# TODO: We probably don't want assets in the image; remove this once we have a proper deployment process which uploads to (e.g.) S3.
-RUN bundle exec rails assets:precompile && rm -fr /app/log
+COPY . .
+RUN bootsnap precompile --gemfile .
+RUN rails assets:precompile && rm -fr log
```

Comment:
is the TODO comment above still valid?

Reply:
Good question :) My take: the question in the TODO is still valid, but I decided not to keep it in here because 1) not sure it's all that helpful having a TODO repeated across 50+ repos and 2) I'm leaning more towards keeping the assets in cos it makes the images quite a bit more flexible (e.g. local e2e testing with minikube/kind/k3s/whatever, makes it simply if we want the option to serve everything from origin as a CDN-is-down [black swan](https://en.wikipedia.org/wiki/The_Black_Swan:_The_Impact_of_the_Highly_Improbable) contingency plan, etc.) and it doesn't add all that much to image size or build time and keeps release and rollout simple and decoupled. Could still be persuaded either way though.